### PR TITLE
Move more setup from Bash to Rust

### PR DESCRIPTION
**Pull request by fitztrev**
Bash script will now compile the binary and then the host will run the binary directly, instead of running `cargo run` via Docker like it did before.

The Rust code is now responsible for creating the repo dirs and cloning.

Goal is to cleanup the bash script and try to limit it to being just a wrapper around various `docker` commands.

This branch on gitpod: https://gitpod.io/#https://github.com/lichess-org/lila-docker/tree/rust

### CODE REVIEW

**Review by trevorbayless — APPROVED** [review]
Great changes and working perfectly! I'd recommend using `if command -v rustup &> /dev/null` instead of `which` though. Not only will it remove the extra output that's generated when `rustup` is missing, but it's also safer. Here's a good write up on it: https://stackoverflow.com/a/677212/4433744

File: command/src/main.rs [hunk 1]
```diff
@@ -37,25 +57,54 @@ enum ComposeProfile {
...
 }
 
 fn main() -> std::io::Result<()> {
+    let args: Vec<String> = std::env::args().collect();
+
+    assert!(args.len() > 1, "Missing command");
+
+    match args[1].as_str() {
+        "setup" => setup(),
+        "gitpod-welcome" => {
```
Comment (kraktus) [inline command/src/main.rs:99]: Code-golf, could be Ok(fn())

File: command/src/main.rs [hunk 2]
```diff
@@ -82,60 +131,83 @@ fn main() -> std::io::Result<()> {
...
+    let repos_to_clone = default_repos
+        .iter()
+        .chain(optional_repos.iter())
+        .collect::<Vec<&String>>();
+
+    for repo in repos_to_clone {
+        let repo_url = format!("https://github.com/{repo}.git");
+        let mut progress = spinner();
+        progress.start(&format!("Cloning {repo}"));
+        let folder = Path::new(&repo).file_name().unwrap();
```
Comment (kraktus) [inline command/src/main.rs:177]: Unwrap can be replaced by try I think here

File: command/src/main.rs [hunk 3]
```diff
@@ -82,60 +131,83 @@ fn main() -> std::io::Result<()> {
...
+        assert!(
+            output.status.success(),
+            "Failed to clone {repo} - {output:?}"
+        );
+
+        progress.stop(format!("Clone {repo} ✓"));
     }
 
+    std::fs::write(".env", config.to_env())?;
+    log::success("Wrote .env")?;
```
Comment (kraktus) [inline command/src/main.rs:205]: Code golf, remove try operator and ok(())

**Pull request merged**